### ISSUE TITLE
exception when execute an AQL query using Web interface

### ISSUE THREAD

**Issue by sebastiancodes** [arangodb/arangodb#2026]
When I'm using the Web Interface to execute a query, I encounter the following behaviour.

```
LET start_end = ["mycollection/9141817", "mycollection/9210025"]

LET vertex1 = start_end[0]
LET vertex2 = start_end[1]

FOR v IN ANY SHORTEST_PATH vertex1 TO vertex2
    GRAPH 'myGraph' 
    RETURN v.name
```

Executes without any issues. However, if try to get rid of the vertex1 and vertex2 assignments, I get an error:

```

LET start_end = ["mycollection/9141817", "mycollection/9210025"]

FOR v IN ANY SHORTEST_PATH start_end[0] TO start_end[1]
    GRAPH 'myGraph' 
    RETURN v.name

```

This results in a warning `Successfully aborted`.

And when I explain:

`Explain: missing variable #28 while planning registers (exception location: /tmp/arangodb-20160822-83950-1x83y8u/ArangoDB-3.0.5/arangod/Aql/ExecutionNode.cpp:1163). Please report this error to arangodb.com (while finalizing)`

**Comment by dothebart** [comment]
Reproduced with an example graph:

    var examples = require("@arangodb/graph-examples/example-graph.js");
    var graph = examples.loadGraph("traversalGraph");
    db._explain(`
    LET start_end = ["circles/a", "circles/d"]
    FOR v IN ANY SHORTEST_PATH start_end[0] TO start_end[1]
        GRAPH 'traversalGraph' 
        RETURN v.name
    `)

**Comment by jsteemann** [comment]
Fixed in 3.0.6 and devel.
The issue occurred when the start or end vertices for the shortest path were non-constant expressions that required extra calculations.

**Comment by dothebart** [comment]
3.0.10 is available for download now.
